src/custom/state/orders/reducer.ts

```diff
@@ -108,6 +109,59 @@ export default createReducer(initialState, (builder) =>
       delete state[chainId].expired[id]
       delete state[chainId].cancelled[id]
     })
+    .addCase(addOrUpdateOrdersBatch, (state, action) => {
```

Comment:
This might be over-complicated. Suggestions to improve?

Reply:
i wouldn't do all of the "business logic" inside the reducer, may be better to split this into logic elsewhere no?